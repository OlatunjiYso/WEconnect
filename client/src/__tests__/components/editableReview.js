import React from "react";

import EditableReview from '../../components/editableReview';

describe('All tests for editable review component', () => {
const mockEdit = jest.fn();
const mockUpdate = jest.fn();
  const props = {
    key: 0,
    review: {},
    businessId: 1,
    updateReview: mockUpdate,
    makeEditable: mockEdit,
  };
    it('should be defined', () => {
      expect(EditableReview).toBeDefined();
    });
    it('should render the banner', () => {
      const { wrapper } = shallow(<EditableReview {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
    it("should contain a form for editing review", () => {
        const wrapper = shallow(<EditableReview {...props} />);
        expect(wrapper.find("form").length).toEqual(1);
      });
      it("should call a mock funtion when cancel button is clicked", () => {
        const wrapper = shallow(<EditableReview {...props} /> );
        const cancelButton= wrapper.find('#cancelReviewUpdate');
        cancelButton.simulate( 'click', {
            preventDefault: () => {
            }
           });
        expect(mockEdit).toHaveBeenCalled();
      });
      it("should call a mock funtion when update button is clicked", () => {
        const wrapper = shallow(<EditableReview {...props} /> );
        const event ={
               preventDefault: () => {}
             }
        wrapper.instance().handleSubmit(event);
        expect(mockUpdate).toHaveBeenCalled();
      });
});
