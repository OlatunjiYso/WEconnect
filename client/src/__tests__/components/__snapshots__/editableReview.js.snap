// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All tests for editable review component should render the banner 1`] = `undefined`;
